src/main/scala/edu/twitter/SentimentAnalyzer.scala

```diff
@@ -32,15 +33,27 @@ object SentimentAnalyzer extends App {
   val modelService = new ModelService(new GradientBoostingBuilder(sc))
   modelService.start()
 
+  val indexHandler = new IndexHandler
+  val result = indexHandler.create("twitter", "sentiment")
+
+  var indexName = ""
+
+  result match {
+    case Left(failure) => terminate()
```

Comment:
Have you tested this branch, I don't know what will happen if we closed the `sparkContext` and continued the execution. would we still train the models?. I guess it would be better if we hard fail and stop the execution.

 `failure` is not used, consider using `_` :).

Reply:
Good point I will change it.